features/apt_messages.feature

```diff
@@ -374,26 +331,295 @@ Feature: APT Messages
+        """
+        {
+          "messages": [
+            {
+              "begin": "$behave_var{today -3}",
+              "end": "$behave_var{today +1}",
+              "lines": [
+                "one"
+              ]
+            }
+          ]
+        }
+        """
+        When I wait `1` seconds
+        When I run `pro refresh messages` with sudo
+        When I run shell command `rm -f /var/lib/ubuntu-advantage/messages/apt-pre*` with sudo
+        When I run `apt upgrade` with sudo
+        Then I will see the following on stdout
+        """
+        Reading package lists...
+        Building dependency tree...
+        Reading state information...
+        Calculating upgrade...
         #
-        # News about significant security updates, features and services will
-        # appear here to raise awareness and perhaps tease /r/Linux ;\)
-        # Use 'pro config set apt_news=false' to hide this and future APT news\.
+        # one
         #
+        0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
+        """
+
+        # begin >30 days ago ignored, even if end is set to future
+        When I create the file `/var/www/html/aptnews.json` on the `apt-news-server` machine with the following:
+        """
+        {
+          "messages": [
+            {
+              "begin": "$behave_var{today -31}",
+              "end": "$behave_var{today +1}",
+              "lines": [
+                "one"
+              ]
+            }
+          ]
+        }
+        """
+        When I wait `1` seconds
+        When I run `pro refresh messages` with sudo
+        When I run shell command `rm -f /var/lib/ubuntu-advantage/messages/apt-pre*` with sudo
+        When I run `apt upgrade` with sudo
+        Then I will see the following on stdout
+        """
+        Reading package lists...
+        Building dependency tree...
+        Reading state information...
+        Calculating upgrade...
+        0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
+        """
+
+        # begin in future
+        When I create the file `/var/www/html/aptnews.json` on the `apt-news-server` machine with the following:
+        """
+        {
+          "messages": [
+            {
+              "begin": "$behave_var{today +1}",
+              "lines": [
+                "one"
+              ]
+            }
+          ]
+        }
+        """
+        When I wait `1` seconds
+        When I run `pro refresh messages` with sudo
+        When I run shell command `rm -f /var/lib/ubuntu-advantage/messages/apt-pre*` with sudo
+        When I run `apt upgrade` with sudo
+        Then I will see the following on stdout
+        """
+        Reading package lists...
+        Building dependency tree...
+        Reading state information...
+        Calculating upgrade...
+        0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
```

Comment:
@orndorffgrant Sorry if I am missing something, but can we add a test for the case where we have the contract expired message as well

Reply:
Nevermind, you already have a PR for [that](https://github.com/canonical/ubuntu-advantage-client/pull/2339)